cmd/logs_test.go

```diff
@@ -0,0 +1,306 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reanahub/reana-client-go/client/operations"
+	"reanahub/reana-client-go/utils"
+	"reflect"
+	"testing"
+)
+
+var logsPathTemplate = "/api/workflows/%s/logs"
+
+func TestLogs(t *testing.T) {
+	workflowName := "my_workflow"
+	logsTemplate := `{
+			"workflow_logs": "workflow logs",
+			"job_logs": {
+				"1": {
+					"workflow_uuid": "%s",
+					"job_name": "%s",
+					"compute_backend": "Kubernetes",
+					"backend_job_id": "backend1",
+					"docker_img": "docker1",
+					"cmd": "ls",
+					"status": "finished",
+					"logs": "%s",
+					"started_at": "2022-07-20T12:09:09",
+					"finished_at": "2022-07-20T19:09:09"
+				},
+				"2": {
+					"workflow_uuid": "workflow_2",
+					"job_name": "job2",
+					"compute_backend": "Slurm",
+					"backend_job_id": "backend2",
+					"docker_img": "docker2",
+					"cmd": "cd folder",
+					"status": "running",
+					"logs": "workflow 2 logs",
+					"started_at": "2022-07-21T12:09:09",
+					"finished_at": "2022-07-21T19:09:09"
+				}
+			},
+			"engine_specific": "engine logs"
+		}`
+	successResponseRaw, _ := json.Marshal(operations.GetWorkflowLogsOKBody{
+		Logs:         fmt.Sprintf(logsTemplate, "workflow_1", "job1", "workflow 1 logs"),
+		User:         "user",
+		WorkflowID:   "my_workflow_id",
+		WorkflowName: "my_workflow",
+	})
```

Comment:
The response was getting a mess so I decided to marshal the body struct instead of writing a string. Should I do this in other places too?

Reply:
Do you mean in `logs` or everywhere else? Hmm, seeing textual "success responses" for commands like `info` in the test file actually feels nice to me; we can see IRL about this change.

Reply:
I meant everywhere else. If you think it's better as text, then we can leave it like that